OpenEdXMobile/src/main/java/org/edx/mobile/http/serialization/OkCallback.java

```diff
@@ -0,0 +1,87 @@
+package org.edx.mobile.http.serialization;
+
+import android.support.annotation.NonNull;
+
+import org.edx.mobile.http.HttpResponseStatusException;
+
+import java.io.IOException;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.Response;
+
+/**
+ * Abstract implementation of OkHttp's
+ * {@link Callback} interface, which provides (and
+ * delegates to) a simpler interface for subclasses,
+ * stripping out unnecessary parameters, and
+ * redirecting all responses with error codes to the
+ * failure callback method.
+ */
+public abstract class OkCallback implements Callback {
```

Comment:
The MVP implementation in the profiles module handle errors itself, so I had defined an abstract base Retrofit `Callback` implementation that only changed the callback semantics according to the pattern we're using (treating HTTP errors as standard failures), without any further logic. It makes sense to have an equivalent implementation of OkHttp's callback implementation as well, in case we need to use it in the same framework.